sdk/python/feast/cli.py

```diff
@@ -151,6 +152,11 @@ def data_source_describe(ctx: click.Context, name: str):
         print(e)
         exit(1)
 
+    warnings.warn(
+        "Describing data sources will only work properly if all data sources have names or table names specified. "
+        "Starting Feast 0.21, data source unique names will be required to encourage data source discovery.",
```

Comment:
Do you think it makes sense to start wrapping these warnings in function?

Something like 

```
feast_deprecation_warning("... Starting Feast {version}, data source unique names will be required to encourage data source discovery", "0.21.0", RuntimeWarning)

def feast_deprecation_warning(message, version, warning_type):
  warnings.warn(message.format(version=version), warning_type)
```

That way we have a single entrypoint for all these deprecation warnings and can easily track ones that are due for actual deprecation once we cross the version.

Reply:
@woop do you feel strongly about the above? I thought about this for a bit, and I don't think it actually makes it much easier to track deprecation. Currently I can just search the codebase for `DeprecationWarning` or for `0.21` to find upcoming things being deprecated; if we introduced `feast_deprecation_warning`, wouldn't we still need to search for `feast_deprecation_warning`? With that being said, I think if we could standardize our error messages somehow, it might still be useful.

Reply:
I agree with Felix here, the feast deprecation warning just redirects the warning call, it does standardize the versioning though.